compiler/ir/ir.tree/src/org/jetbrains/kotlin/ir/util/AdditionalIrUtils.kt

```diff
@@ -85,11 +85,21 @@ fun IrSimpleFunction.overrides(other: IrSimpleFunction): Boolean {
 private val IrConstructorCall.annotationClass
     get() = this.symbol.owner.constructedClass
 
+val IrClass.packageFqName
+    get() = parent.getPackageFragment()?.fqName
+
+tailrec fun IrDeclarationWithName.hasEqualFqName(fqName: FqName): Boolean {
```

Comment:
This function doesn't seem `tailrec` to me

Reply:
Also, I propose to refactor a bit:
```
fun IrDeclarationWithName.hasEqualFqName(fqName: FqName): Boolean =
    name == fqName.shortName() && when (val parent = parent) {
        is IrPackageFragment -> parent.fqName == fqName.parent()
        is IrDeclarationWithName -> parent.hasEqualFqName(fqName.parent())
        else -> false
    }
```

Reply:
> This function doesn't seem tailrec to me